db/migrate/20241030154245_create_life_cycles.rb

```diff
@@ -0,0 +1,39 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class CreateLifeCycles < ActiveRecord::Migration[7.1]
+  def change
+    create_table :life_cycles do |t|
+      t.string :type
+      t.string :name
+      t.belongs_to :color, type: :int
```

Comment:
Good point! Not sure, this is how it was defined for the other relations in the migrations and I just followed this pattern:
https://github.com/opf/openproject/blob/0ae0349d9bd3f305bcabe2892937c4b3f4417dd4/db/migrate/20180504144320_add_color_to_statuses_and_enumerations.rb#L31-L37
https://github.com/opf/openproject/blob/0ae0349d9bd3f305bcabe2892937c4b3f4417dd4/db/migrate/tables/types.rb#L40

They are old and I'm not sure if there was a specific reason to do this.
Looking through the structure sql, we have a few more examples of `*_id` columns defined as `integer` when their corresponding primary key columns are bigint.
This could cause a problem at some point I guess, I logged a [code maintenance work package](https://community.openproject.org/wp/59476) to solve the other examples.